tests/test_cityscapes_format.py

```diff
@@ -0,0 +1,379 @@
+                        Mask(image=np.array([[0, 0, 1, 0, 1]]), label=2, id=2,
+                            attributes={'is_crowd': True}),
+                    ]),
+                ])
+
+        with TestDir() as test_dir:
+            self._test_save_and_load(TestExtractor(),
+                partial(CityscapesConverter.convert, label_map='cityscapes',
+                save_images=True), test_dir)
+
+    def test_can_save_cityscapes_dataset_without_frame_and_sequence(self):
+        class TestExtractor(TestExtractorBase):
+            def __iter__(self):
+                return iter([
+                    DatasetItem(id='justcity', subset='test',
+                        image=np.ones((1, 5, 3)), annotations=[
+                        Mask(image=np.array([[1, 0, 0, 1, 1]]), label=3, id=3,
+                            attributes={'is_crowd': True}),
+                        Mask(image=np.array([[0, 1, 1, 0, 0]]), label=24, id=1,
+                            attributes={'is_crowd': False}),
+                    ]),
+                ])
+        with TestDir() as test_dir:
+            self._test_save_and_load(TestExtractor(),
+                partial(CityscapesConverter.convert, label_map='cityscapes',
+                save_images=True), test_dir)
+
+    def test_can_save_dataset_with_cyrillic_and_spaces_in_filename(self):
+        class TestExtractor(TestExtractorBase):
+            def __iter__(self):
+                return iter([
+                    DatasetItem(id='кириллица с пробелом',
+                       image=np.ones((1, 5, 3)), annotations=[
+                         Mask(image=np.array([[1, 0, 0, 1, 1]]), label=3, id=3,
+                             attributes={'is_crowd': True}),
+                         Mask(image=np.array([[0, 1, 1, 0, 0]]), label=24, id=1,
+                             attributes={'is_crowd': False}),
+                    ]),
+                ])
+
+        with TestDir() as test_dir:
+            self._test_save_and_load(TestExtractor(),
+                partial(CityscapesConverter.convert, label_map='cityscapes',
+                save_images=True), test_dir)
+
+    def test_can_save_cityscapes_dataset_with_strange_id(self):
+        class TestExtractor(TestExtractorBase):
+            def __iter__(self):
+                return iter([
+                    DatasetItem(id='a/b/1', subset='test',
+                        image=np.ones((1, 5, 3)), annotations=[
+                        Mask(image=np.array([[1, 0, 0, 1, 1]]), label=3, id=3,
+                            attributes={'is_crowd': True}),
+                        Mask(image=np.array([[0, 1, 1, 0, 0]]), label=24, id=1,
+                            attributes={'is_crowd': False}),
+                    ]),
+                ])
+
+        class DstExtractor(TestExtractorBase):
+            def __iter__(self):
+                return iter([
+                    DatasetItem(id='a_b_1', subset='test',
```

Comment:
Let's rethink "strange id" to be a "relative path to a custom image in a custom dataset" and try to keep the original file layout.

Reply:
Rewritten